### PR TITLE
[ci] #4459: New iroha2-stable images release tags

### LINKED ISSUE #4459: Release publishing
On every tag that starts with `v2*` following docker images should be built:
1. `iroha:*` with alias `iroha:2.0.0-latest`
2. `iroha:*-profiling`

### Example

1. a tag **`v2.0.0-pre-rc-21`** is published then:
2. we build **`iroha:2.0.0-pre-rc-21`** and **`iroha:2.0.0-pre-rc-21-profiling`** docker images
3. we update docker alias **`iroha:2.0.0-latest`** to reference to `iroha:2.0.0-pre-rc-21`

notice that tags will always be prefixed with `v2` or `v1` but docker tags will strip the `v` prefix

### Alternate suggestions

please provide if you have

### PR DESCRIPTION
**Pull request by BAStos525**
### Description of the Change
Modify `iroha2-stable` and `iroha2-profiling` images tags regarding release publishing.

### Benefits
Closes #4459

### CODE REVIEW

File: .github/workflows/iroha2-profiling-image.yml [hunk 1]
```diff
@@ -31,6 +18,12 @@ jobs:
       image: hyperledger/iroha2-ci:nightly-2024-04-18
     steps:
       - uses: actions/checkout@v4
+      - name: Get the release tag
+        run: |
+          RELEASE_VERSION=${{ github.event.release.tag_name }}
+          PREFIX='v'
+          TAG=${RELEASE_VERSION#$PREFIX}
```
Comment (mversic) [inline .github/workflows/iroha2-profiling-image.yml:25]: does this strip `v` prefix? from `RELEASE_VERSION`?

File: .github/workflows/iroha2-release.yml [hunk 2]
```diff
@@ -34,9 +32,8 @@ jobs:
           load: true
           file: Dockerfile
           tags: |
-            hyperledger/iroha2:${{ env.TAG }}
-            hyperledger/iroha2:${{ env.TAG }}-${{ env.RELEASE }}
-            docker.soramitsu.co.jp/iroha2/iroha2:${{ env.TAG }}-${{ env.RELEASE }}
+            hyperledger/iroha:${{ env.TAG }}
+            docker.soramitsu.co.jp/iroha2/iroha:${{ env.TAG }}
```
[the remark this answers is not in the document]
Comment (BAStos525) [reply .github/workflows/iroha2-release.yml:36]: This is the organization of the image structure in the registry and it is correct. You don't need to touch it.

File: .github/workflows/iroha2-profiling-image.yml (hunk 1 above)
Comment (BAStos525) [reply .github/workflows/iroha2-profiling-image.yml:25]: Yes, I checked in bash

File: .github/workflows/iroha2-profiling-image.yml [hunk 3]
```diff
@@ -47,20 +41,20 @@ jobs:
...
+      - name: Build and push IROHA2_PROFILE image
         uses: docker/build-push-action@v5
         if: always()
         with:
           push: true
           tags: |
-            hyperledger/iroha2:${{ github.event.inputs.IROHA2_IMAGE_TAG }}-${{ github.event.inputs.IROHA2_IMAGE_RELEASE }}-profiling
-            docker.soramitsu.co.jp/iroha2/iroha2:${{ github.event.inputs.IROHA2_IMAGE_TAG }}-${{ github.event.inputs.IROHA2_IMAGE_RELEASE }}-profiling
+            hyperledger/iroha:${{ env.TAG }}-IROHA2_PROFILE
+            docker.soramitsu.co.jp/iroha2/iroha:${{ env.TAG }}-IROHA2_PROFILE
```
Comment (mversic) [inline .github/workflows/iroha2-profiling-image.yml:51]:
```suggestion
            hyperledger/iroha:${{ env.TAG }}-$IROHA2_PROFILE
            docker.soramitsu.co.jp/iroha2/iroha:${{ env.TAG }}-$IROHA2_PROFILE
```

I'm not sure if you need `$` here

**Review by mversic — DISMISSED** [review]

**Review by mversic — APPROVED** [review]

**Review by Erigara — APPROVED** [review]

**Pull request merged**